### PR TITLE
Add support for selecting the node to remove upon scale down

### PR DESCRIPTION
**Pull request by luis5tb**
This PR adds a new option to the CR spec to allow the user to both
select the node to scale down and the draining process. The Spec
field added is:
  spec:
    ...
    nodesToDelete:
    - name: node-name (e.g., worker-3)
      drain: true|false
    - name: ...

Note the drain is optional and considered false if not set. The
functionality behind the drain will come in follow up PRs. The idea
behind is that if drain is set to false, the user will need to drain
the node and the operator will wait for it to delete the node. If
it is set to true, and automatic draining will be triggered (with
no guarantees of finishing).

More than one node can be added to the nodesToDelete list, and what
the operator does is to add the required annotations into their
associated machine objects, so that those are prioritized for removal
once the scale down is triggered. If nothing stated, the operator
will choose the newest machine.

In addition this PR adds two extra things:
- Reconcile every 10 mins, regardless of changes to the CR or the
associated objects being watched
- Ensure if machineset replicas are adjusted, the CR get updated
with that information

### CODE REVIEW

File: deploy/crds/compute-node.openstack.org_computenodeopenstacks_crd.yaml [hunk 1]
```diff
@@ -63,6 +63,22 @@ spec:
...
               description: Kubernetes service cluster IP
               type: string
+            nodesToDelete:
+              description: Nodes to delete upon scale down
+              items:
+                description: NodeToDelete defines the name of the node to delete and
+                  if automatic drain is needed
+                properties:
+                  drain:
+                    description: Automatic draining of the node
```
Comment (stuggi) [inline deploy/crds/compute-node.openstack.org_computenodeopenstacks_crd.yaml:73]: with draining you mean migrate instances off the compute node, or all tasks run on a scale down?

Comment (stuggi) [reply deploy/crds/compute-node.openstack.org_computenodeopenstacks_crd.yaml:73]: I mean do we want to skip all tasks when draining is disabled or just the auto migration?

Comment (luis5tb) [reply deploy/crds/compute-node.openstack.org_computenodeopenstacks_crd.yaml:73]: auto-migration only!

Comment (stuggi) [reply deploy/crds/compute-node.openstack.org_computenodeopenstacks_crd.yaml:73]: as discussed this is a switch to enable/disable libe migration

**Pull request merged**